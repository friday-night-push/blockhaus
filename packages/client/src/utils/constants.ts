export const HOST = 'http://friday-night-push-blockhau-40.ya-praktikum.tech';
export const SERVER_HOST_URL = `${HOST}:3001`;
export const API_URL = `${SERVER_HOST_URL}/api/v1`;
export const YANDEX_URL = `${API_URL}/proxy`;

// OAuth
export const REDIRECT_URL = `${HOST}:3000`;

export const RESOURCE_URL = YANDEX_URL + '/resources';

export enum PAGE_ROUTES {
  MENU = '/',
  SIGN_IN = '/sign-in',
  SIGN_UP = '/sign-up',
  CHANGE_PASSWORD = '/change-password',
  RECOVER_PASSWORD = '/recover-password',
  PROFILE = '/profile',
  GAME = '/game',
  GAME_PAUSE = '/game-pause',
  GAME_RESULT = '/game-result',
  GAME_SETUP = '/game-setup',
  GAME_DIFFICULT = '/game-difficult',
  LEADER_BOARD = '/leader-board',
  ERROR_BOUNDARY = '/error-boundary',
  FORUM = '/forum',
  FORUM_TOPIC = '/forum/:topicId/:page?',
}
