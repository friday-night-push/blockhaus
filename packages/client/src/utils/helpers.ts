// Отдельный класс исключений для ESLint

export default class Helpers {
  public static Log(level: string, ...args: any): any {
    switch (level) {
      case 'INFO':
        // eslint-disable-next-line no-console
        console.info(...args);
        break;
      case 'LOG':
        // eslint-disable-next-line no-console
        console.log(...args);
        break;
      case 'ERROR':
        // eslint-disable-next-line no-console
        console.error(...args);
        break;
      default:
        break;
    }
  }
}
