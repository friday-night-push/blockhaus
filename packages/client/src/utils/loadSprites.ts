export function loadSprites(
  spritesPath: Record<string, string> = {}
): Promise<Record<string, HTMLImageElement>> {
  return Promise.all(
    Object.values(spritesPath).map(path => loadImage(path))
  ).then(sprites => {
    const keys = Object.keys(spritesPath);
    const map = keys.map((key, index) => [key, sprites[index]]);
    return Object.fromEntries(map);
  }) as Promise<Record<string, HTMLImageElement>>;
}

function loadImage(src: string): Promise<HTMLImageElement> {
  return new Promise<HTMLImageElement>((resolve, reject) => {
    const img = new Image();
    img.onload = () => resolve(img);
    img.onerror = reject;
    img.src = src;
  });
}
