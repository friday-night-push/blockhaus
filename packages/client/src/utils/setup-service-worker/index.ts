export * from './setup-service-worker';
