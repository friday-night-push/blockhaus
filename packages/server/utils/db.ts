import { Sequelize } from 'sequelize-typescript';

const { POSTGRES_USER, POSTGRES_PASSWORD, POSTGRES_DB, POSTGRES_PORT } =
  process.env;

export const initPostgres = async (): Promise<Sequelize | null> => {
  try {
    const sequelize = new Sequelize({
      host: 'localhost',
      port: Number(POSTGRES_PORT),
      username: POSTGRES_USER,
      password: POSTGRES_PASSWORD,
      database: POSTGRES_DB,
      dialect: 'postgres',
      logging: false,
    });
    await sequelize.authenticate();
    await sequelize.sync({ force: true });
    console.log('All models were synchronized successfully.');
    return sequelize;
  } catch (e) {
    console.error(e);
  }
  return null;
};
