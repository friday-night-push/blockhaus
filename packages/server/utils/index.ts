export * from './init-postgres';
export * from './is-dev';
export * from './logger';
