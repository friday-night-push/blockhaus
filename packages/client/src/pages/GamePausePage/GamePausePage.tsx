import { Menu } from '@gravity-ui/uikit';
import { useNavigate } from 'react-router-dom';

import { Container } from 'src/components/atoms/Container';
import type { MenuItemProps } from 'src/components/atoms/MenuItem';
import { MenuItem } from 'src/components/atoms/MenuItem';
import { Page } from 'src/components/organisms/Page';
import { PAGE_ROUTES } from 'src/utils/constants';

export const GamePausePage = () => {
  const navigate = useNavigate();

  const stopGame = () => {
    localStorage.removeItem('scores');
    localStorage.removeItem('field');
    localStorage.removeItem('time');
    localStorage.removeItem('type');
    localStorage.removeItem('difficult');
    navigate(PAGE_ROUTES.MENU);
  };

  const MENU_ITEMS: MenuItemProps[] = [
    { label: 'resume', href: PAGE_ROUTES.GAME },
    { label: 'stop', onClick: stopGame, theme: 'danger' },
  ];

  return (
    <Page>
      <Menu size={'xl'}>
        <Container direction={'column'} alignItems={'center'}>
          {MENU_ITEMS.map(item => (
            <MenuItem key={item.label} {...item} />
          ))}
        </Container>
      </Menu>
    </Page>
  );
};
