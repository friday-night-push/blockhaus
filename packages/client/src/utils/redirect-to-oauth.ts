import { REDIRECT_URL } from './constants';

export const redirectToOAuth = (serviceId: string) => {
  window.location.replace(
    `https://oauth.yandex.ru/authorize?response_type=code&client_id=${serviceId}&redirect_uri=${REDIRECT_URL}`
  );
};
