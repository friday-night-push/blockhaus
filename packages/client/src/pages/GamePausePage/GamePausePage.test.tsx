import { render } from 'src/utils/tests';

import { GamePausePage } from './GamePausePage';

describe('GamePausePage', () => {
  it('matches snapshot', () => {
    const { container } = render(<GamePausePage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
