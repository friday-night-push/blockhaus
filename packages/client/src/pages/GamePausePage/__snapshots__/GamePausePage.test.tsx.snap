// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamePausePage matches snapshot 1`] = `
<section
  class="g-box g-flex"
  style="width: auto; min-height: 100vh; column-gap: 12px; row-gap: 12px; justify-content: center;"
>
  <section
    class="g-box g-flex"
    style="width: 100%; min-height: 100vh; max-width: 380px; flex-direction: column; column-gap: 32px; row-gap: 32px; align-items: center; justify-content: center;"
  >
    <span
      class="g-text g-text_variant_header-1"
      style="text-align: center;"
    >
      PAUSE THE GAME
    </span>
    <ul
      class="g-menu g-menu_size_xl"
      role="menu"
    >
      <section
        class="g-box g-flex"
        style="flex-direction: column; column-gap: 12px; row-gap: 12px; align-items: center;"
      >
        <li
          class="g-menu__list-item"
        >
          <div
            class="g-menu__item g-menu__item_theme_normal g-menu__item_interactive"
            role="menuitem"
            style="min-height: 50px;"
            tabindex="0"
          >
            <div
              class="g-menu__item-content"
            >
              <span
                class="g-text g-text_variant_display-2"
              >
                resume the game
              </span>
            </div>
          </div>
        </li>
        <li
          class="g-menu__list-item"
        >
          <div
            class="g-menu__item g-menu__item_theme_normal g-menu__item_interactive"
            role="menuitem"
            style="min-height: 50px;"
            tabindex="0"
          >
            <div
              class="g-menu__item-content"
            >
              <span
                class="g-text g-text_variant_display-2"
              >
                cancel the game
              </span>
            </div>
          </div>
        </li>
      </section>
    </ul>
    <section
      class="g-box g-flex"
      style="column-gap: 12px; row-gap: 12px; align-items: center;"
    >
      <span
        class="g-text g-text_variant_code-2"
      >
        Made in 2024 by
      </span>
      <a
        class="g-link g-link_view_normal"
        href="https://github.com/friday-night-push/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
        />
      </a>
      <span
        class="g-text g-text_variant_code-2"
      >
        at
      </span>
      <a
        class="g-link g-link_view_normal"
        href="https://practicum.yandex.ru/middle-frontend/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <img
          alt=""
        />
      </a>
    </section>
  </section>
</section>
`;
